tests/ValueOfTest.php

```diff
@@ -212,6 +280,23 @@ function getValue() {
                 ',
                 'error_message' => 'InvalidReturnStatement'
             ],
+            'valueOfUnitEnum' => [
+                'code' => '<?php
+                    enum Foo
+                    {
+                        case Foo;
+                        case Bar;
+                    }
+
+                    /** @param value-of<Foo> $arg */
+                    function foobar(string $arg): void {}
+                ',
+                // TODO turn this into an InvalidDocblock with a better error message. This is difficult because it
+                // has to happen after scanning has finished, otherwise the class might not have been scanned yet.
```

Comment:
I'd like to get this to work, but unless there's a single point of analysis for docblocks after scanning has finished I'm not sure how, so I figure it can wait.